oligo_designer_toolsuite/IO/_database.py

```diff
@@ -1,139 +1,214 @@
+    def __init__(self) -> None:
 
+        self.species = "human"
+        self.genome_assembly = "GRCh38"
+        self.annotation_release = "current"
 
     def read_DB(self, file_DB):
         if os.path.exists(file_DB):
             if data_parser.check_fasta_format(file_DB):
                 self.file_DB = file_DB
             else:
-                raise ValueError('Database has incorrect format!')
+                raise ValueError("Database has incorrect format!")
         else:
-            raise ValueError('Database file does not exist!')
-        pass
-
-
-    def download_annotation(self, species, genome_assembly, annotation_source, annotation_release, dir_output):
+            raise ValueError("Database file does not exist!")
+
+    def download_annotation(
+        self,
+        species,
+        genome_assembly,
+        annotation_source,
+        annotation_release,
+        dir_output,
+    ):
         if species is None:
-            warnings.warn(f'No species defined. Using default species {self.species}!')
+            warnings.warn(f"No species defined. Using default species {self.species}!")
             species = self.species
 
         if genome_assembly is None:
-            warnings.warn(f'No genome assembly defined. Using default assembly {self.genome_assembly}!')
+            warnings.warn(
+                f"No genome assembly defined. Using default assembly {self.genome_assembly}!"
+            )
             genome_assembly = self.genome_assembly
 
         if annotation_release is None:
-            warnings.warn(f'No ncbi annotation release defined. Using default release {self.annotation_release}!')
+            warnings.warn(
+                f"No ncbi annotation release defined. Using default release {self.annotation_release}!"
+            )
             annotation_release = self.annotation_release
 
-        if annotation_source == 'NCBI':
+        if annotation_source == "NCBI":
             FTP = ftp_loader.FTPLoaderNCBI(species, genome_assembly, annotation_release)
-        elif annotation_source == 'ensemble'
-            FTP = ftp_loader.FtpLoaderEnsemble(species, genome_assembly, annotation_release)
-        
+        elif annotation_source == "ensemble":
+            FTP = ftp_loader.FtpLoaderEnsemble(
+                species, genome_assembly, annotation_release
+            )
+
         file_annotation = FTP.download_gtf(dir_output)
         file_sequence = FTP.download_fasta(dir_output)
 
         return file_annotation, file_sequence
 
-    def _get_gene_transcript(self, file_sequence, file_annotation, blockSize, dir_output):
+    def _get_gene_transcript(
+        self, file_sequence, file_annotation, blockSize, dir_output
+    ):
         pass
 
-    '''
+    """
     def _get_gene(self, file_sequence, file_annotation, dir_output):
         pass
 
     def _get_gene_CDS(self, file_sequence, file_annotation, blockSize, dir_output):
         pass
-    '''
-
+    """
 
 
```

Comment:
For the following, it makes more sense to me to have these classes:

- NcbiDbCreator
- EnsembleDbCreator
- CustomDbCreator

Each would implement two functions:
- createReferenceDB
- createOligoDB(OligoClass)

And the oligo class would be outside of IO, and would contain Oligo related functions
I think we should discuss this @lisa-sousa and @francesco-campi :)